packages/contracts-bedrock/test/dispute/FaultDisputeGame.t.sol

```diff
@@ -96,6 +96,12 @@ contract FaultDisputeGame_Init is DisputeGameFactory_Init {
         // Register the game implementation with the factory.
         disputeGameFactory.setImplementation(GAME_TYPE, gameImpl);
         uint256 bondAmount = disputeGameFactory.initBonds(GAME_TYPE);
+
+        // Warp ahead of the game retirement timestamp if needed.
+        if (block.timestamp <= anchorStateRegistry.retirementTimestamp()) {
+            vm.warp(anchorStateRegistry.retirementTimestamp() + 1);
+        }
```

Comment:
Why did we need to dd this if we didn't change the FDG?

Reply:
I changed `proveWithdrawalTransaction` to do this check. We already had the check in `finalizeWithdrawalTransaction` but we didn't have the check in `proveWithdrawalTransaction` which led to a situation where you can prove certain withdrawals that you can't finalize. I believe this was a low-sev finding in a previous audit, resolving it now.